scene/gui/view_panner.cpp

```diff
@@ -92,19 +92,34 @@ bool ViewPanner::gui_input(const Ref<InputEvent> &p_event, Rect2 p_canvas_rect)
 			return false;
 		}
 
+		bool is_zoom_drag_event = mb->get_button_index() == MouseButton::MIDDLE && mb->is_ctrl_pressed();
```

Comment:
I think this implementation makes the code a bit complicated and hard to read. I suggest using an enum with a DRAG_TYPE_ instead. You can check how it's done in the TileMap editors, but basically, the idea would to use something like this:
```
	enum DragType {
		DRAG_TYPE_NONE,
		DRAG_TYPE_PAN,
		DRAG_TYPE_ZOOM,
	};
```
This makes sure all states are mutually exclusive and there's no way the both zoom and pan are possible at the same time.